experiments/mp-game-6/draw.creatures.js

```diff
@@ -116,20 +161,46 @@ function drawTestCreature(stim, creatureInd, scale, roundProps){
             markAsSpecies(id);
             roundProps.selected_test_stim.push(id);
         }
-        console.log(roundProps.selected_test_stim)
     });
 }
   
 function darken(id) {
-    $(id).css({"opacity": 1.0});
+    $(id).css({"opacity":1.0});
+}
+function lighten(id){
+    $(id).css({"opacity":.5});
 }
   
-function showSpeciesIndicator(id, speciesName) {
-    var labelID = id + "_label";
-    $(labelID).css("visibility", "visible");
-  
-    if ($(id).attr("belongs_to_concept") === "true")
-        $(id).css({"border":"2px dashed black"});
+function turnLightsOn(buttonsOn, test1, numButtons, numLights, lightsPrev) {
+    var lightsOn = [];
+    var arrOfButtons = [];
+    for(var i = 1; i<=numButtons; i++){
+        if(buttonsOn.includes("#button" + i)) arrOfButtons.push(true);
+        else arrOfButtons.push(false);
+    }
+    for(var j = 1; j<=numLights; j++){
+        $("#light" + j).css({"opacity":0.5});
+    }
+    var setting1 = new Setting(arrOfButtons)
+    for(var h = 0; h<test1(setting1).length; h++){
+        var id = "#light" + (h+1)
+        if(test1(setting1)[h]){
+            $(id).css({"opacity":1.0});
+            lightsOn.push(id)
+        }
+    }
+    console.log("The following lights are on: " + lightsOn + " And the following lights have been on: " + lightsPrev)
+    for(var k = 0; k<lightsOn.length; k++){
+        if(!lightsPrev.includes(lightsOn[k])){
```

Comment:
Same here, probably better to use a set tor dictionary rather than an array. I.e. the set of lights that have *been* turned on is

```json
{
    'light1': false,
    'light2': false,
    'light3': false
}
```
and you update that as we go along

Reply:
You got rid of `Setting` in the second commit - cool, I would get rid of it in the other places too.

Reply:
Ah, you did!